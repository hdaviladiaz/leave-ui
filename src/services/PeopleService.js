import axios from 'axios';

export default class PeopleService {
  get(id) {
    return axios.get('/api/people/' + id)
    .then(response => {return response.data[0]})
    .catch(error =>	Promise.reject(error));    
  }
}
