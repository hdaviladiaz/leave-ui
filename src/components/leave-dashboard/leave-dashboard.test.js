import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import LeaveDashboard from './leave-dashboard';
import PeopleService from '../../services/peopleService';
import renderer from 'react-test-renderer';
jest.mock('react-dom');

describe('LeaveDashboard', () => {

  beforeEach(function () {
    window.localStorage = storageMock();
  });

  const vivi = {"employeeId":"15576",
                "preferredName":"Viviana Perez",
                "picture":{"url":"https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile"},
                "hireDate":"2013-08-05",
                "homeOffice":{"name":"Quito"},
                "workingOffice":{"name":"Pune"}}

  function storageMock() {
    var storage = {};
    return {
      setItem: function(key, value) {
        storage[key] = value || '';
      },
      getItem: function(key) {
        return key in storage ? storage[key] : null;
      }
    };
  }

  it('renders a snapshot', () => {
    const tree = renderer.create(<LeaveDashboard/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render people information when person is resolved', () => {
    const component = renderer.create(<LeaveDashboard/>);
    component.getInstance().setState({person: vivi});
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render error when person is resolved', () => {
    const component = renderer.create(<LeaveDashboard/>);
    component.getInstance().setState({error: {}});
    expect(component.toJSON()).toMatchSnapshot();
  });

});
