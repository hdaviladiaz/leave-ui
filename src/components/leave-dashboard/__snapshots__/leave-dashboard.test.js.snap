// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveDashboard renders a snapshot 1`] = `
<div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <div
        className="overview"
      >
        <div
          className="col-xs-12"
        >
          <h4>
            22 junio 2017
          </h4>
          <span
            className="text-muted"
          >
            15
             dias disponibles
          </span>
        </div>
        <div
          className="overview-box-container no-padding col-xs-12"
        >
          <div
            className="overview-box col-md-4"
          >
            <div
              className="overview-box-header gray-color"
            >
              Días Disponibles
            </div>
            <div
              className="overview-box-body"
            >
              <div
                className="overview-box-title"
              >
                15
              </div>
            </div>
          </div>
          <div
            className="overview-box col-md-4"
          >
            <div
              className="overview-box-header"
            >
              Última solicitud
            </div>
            <div
              className="overview-box-body"
            >
              <div
                className="overview-box-title"
              >
                 8 días
              </div>
              <div
                className="text-muted"
              >
                15-06-2017
              </div>
            </div>
          </div>
          <div
            className="overview-box last overview-box-center col-md-4"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Solicitar Ahora
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="leave-request text-center panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <img
        alt=""
        className="back-image"
        src="/img/back.png"
      />
      <h4
        className="gray-color"
      >
        No has solicitado vacaciones
      </h4>
    </div>
  </div>
</div>
`;

exports[`LeaveDashboard should render error when person is resolved 1`] = `
<div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <div
        className="overview"
      >
        <div
          className="col-xs-12"
        >
          <h4>
            22 junio 2017
          </h4>
          <span
            className="text-muted"
          >
            15
             dias disponibles
          </span>
        </div>
        <div
          className="overview-box-container no-padding col-xs-12"
        >
          <div
            className="overview-box col-md-4"
          >
            <div
              className="overview-box-header gray-color"
            >
              Días Disponibles
            </div>
            <div
              className="overview-box-body"
            >
              <div
                className="overview-box-title"
              >
                15
              </div>
            </div>
          </div>
          <div
            className="overview-box col-md-4"
          >
            <div
              className="overview-box-header"
            >
              Última solicitud
            </div>
            <div
              className="overview-box-body"
            >
              <div
                className="overview-box-title"
              >
                 8 días
              </div>
              <div
                className="text-muted"
              >
                15-06-2017
              </div>
            </div>
          </div>
          <div
            className="overview-box last overview-box-center col-md-4"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Solicitar Ahora
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="leave-request text-center panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <img
        alt=""
        className="back-image"
        src="/img/back.png"
      />
      <h4
        className="gray-color"
      >
        No has solicitado vacaciones
      </h4>
    </div>
  </div>
</div>
`;

exports[`LeaveDashboard should render people information when person is resolved 1`] = `
<div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <div
        className="overview"
      >
        <div
          className="col-xs-12"
        >
          <h4>
            22 junio 2017
          </h4>
          <span
            className="text-muted"
          >
            15
             dias disponibles
          </span>
        </div>
        <div
          className="overview-box-container no-padding col-xs-12"
        >
          <div
            className="overview-box col-md-4"
          >
            <div
              className="overview-box-header gray-color"
            >
              Días Disponibles
            </div>
            <div
              className="overview-box-body"
            >
              <div
                className="overview-box-title"
              >
                15
              </div>
            </div>
          </div>
          <div
            className="overview-box col-md-4"
          >
            <div
              className="overview-box-header"
            >
              Última solicitud
            </div>
            <div
              className="overview-box-body"
            >
              <div
                className="overview-box-title"
              >
                 8 días
              </div>
              <div
                className="text-muted"
              >
                15-06-2017
              </div>
            </div>
          </div>
          <div
            className="overview-box last overview-box-center col-md-4"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Solicitar Ahora
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="leave-request text-center panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <img
        alt=""
        className="back-image"
        src="/img/back.png"
      />
      <h4
        className="gray-color"
      >
        No has solicitado vacaciones
      </h4>
    </div>
  </div>
</div>
`;
